### DISCUSSION TITLE
Support Windows 11 ARM

### DISCUSSION THREAD

**Discussion by codezi** [valinet/ExplorerPatcher#1045] [Ideas]
Hi,
First of all, thank you very much for creating Explorer Patcher.
I am using Windows 11 ARM with Parallels Desktop for my Macbook. And Explorer Patcher doesn't work (and I know at this time you don't have a plan for the ARM version).
But, hopefully, someday in the future, you support its work on Windows 11 ARM.
Thank you very much.

**Comment by valinet** [comment]
Hi

Thanks for the kind words. The reality is, the more I work on this, the less likely an ARM64 port is going to be. More specialized tweaks require patching instructions directly, so the amount of work for supporting 2 architectures would be too big. Or like, not necessarily the initial work, but maintaining that would be a pain; as if maintaining EP in its current state isn't. This, so as not to mention the usual hiccups, like not having hardware to test on etc.

Anyway, again, thanks for the good words and for checking this out.

**Comment by foorschtbar** [comment]
I have also a Lenovo ARM machine here, too bad that there is no build for Windows on ARM :(

@valinet: I understand you completely, for the "test hardware" is a solution. Oracle offers [free ARM64 machines](https://www.oracle.com/cloud/free/) (Ampere A1)

ARM is the future :)

**Reply by valinet** [reply]
Maybe the future, but definitely not the present. At the moment, the lost productivity due to the architecture switch negates any other benefit for me. Not to mention, an architecture change is an architecture change. I have a ton of software I run that is x86-only - the hassle of working around that is a pain.

**Comment by gittimmys** [comment]
Hi @valinet,

After 39 years of operating in Windows "only" (and sometimes Linux and Unix), I've decided to migrate to Mac OS. However, like you, I have a number of Windows applications that work best in an X86 environment, so, I'm using Parallels VM with Windows 11 ARM. So far only a few of my essential programs are problematic on Windows 11 ARM and I'm working through those issues.

Explorer Patcher is beyond amazing and I categorize it as an "essential" software.

I understand your lack of resources/time to develop Explorer Patcher for multiple operating systems. I'm hoping it might be helpful to add another satisfied customer to your list for your consideration to create or co-create a version of Explorer Patcher for Windows 11 ARM.

**Comment by smnc** [comment]
Well, some fairly big news: the new Prism emulation layer and other compatibility emprovements that are baked in to WoA 24H2 means that ExplorerPatcher _does_ run on Windows 11 on ARM now... with a fairly big caveat that some fairly major things don't work - namely switching back to Windows 10 Start Menu and Task Bar.
![Screenshot 2024-06-15 201957](https://github.com/valinet/ExplorerPatcher/assets/19196582/a545523c-0229-41ee-8603-f9e5149e1aa1)

FWIW, StartAllBack is also working - and it works without issues

**Reply by valinet** [reply]
This has nothing to do with any emulation layer. Emulation layers are not magic wands, an ISA is an ISA. An emulation layer cannot fix someone trying to inject x86 opcodes in an ARM64 binary. I haven't looked personally, but I bet ExplorerPatcher is not injected in `explorer.exe` on your test instance. The fact that the GUI works is not indicative of any of the functionalities actually working (at least those that pertain to patching - ofc, setting registry entries is much less involved). The GUI is hosted in `rundll32`, which on these newer builds probably detects it's been given an x86 DLL to execute and runs it though Prism. But, again, no emulation layer can automagically fix code expecting to patch x86 codes on an ARM64 machine, and ExplorerPatcher does a fair amount of patching.

StartAllBack works because, as far as I remember, actually ships ARM64 binaries, they actually compile and test for ARM64. If you unpack StartAllBack, as far as I remember, there is a native ARM64 DLL which includes the functionality.

**Reply by smnc** [reply]
I get that - didn't mean to suggest otherwise. All I was pointing out is that it was failing to launch outright under the old translation layer, but launches successfully now - for all I know this was an improvement made at some point in a previous release, last time I tried to run EP on WoA was 22H2. With 24H2 on my WoA machine, I thought I'd give it another spin.
I understand that the actual patching won't work without significant effort on your part, and that at present you have no time/inclination to put in that effort (and no access to a WoA device). I greatly appreciate the work you have done on EP, and use it on my other x64 based machines, I just wanted to post an update for those following the thread.

Also, didn't know StartAllBack offered WoA compatibility, last time I checked, they didn't.

**Reply by gittimmys** [reply]
Good to see continued interest!

**Comment by pyrates999** [comment]
That's because the windows 10 taskbar code and windows 10 start menu code have been removed entirely in that build.

**Reply by valinet** [reply]
Nope, has nothing to do with it. The memory patching that ExplorerPatcher does is x86 ONLY at the moment, it only expects and includes patches for the x86 ISA, and no amount of emulation layer/AI/whatever in the world can automagically fix that.
